### PR TITLE
Kindly review

### PR DESCRIPTION
**Pull request by mosaaleb**

### CODE REVIEW

File: assets/script/library.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+function deleteBook() {
+    let books = getBooks();
+    let index = event.target.dataset.index;
+    books.splice(index, 1);
+    // console.log(myLibrary);
+    localStorage.books = JSON.stringify(books);
+    document.location.reload();
+}
+
+function readBookToggle() {
```
Comment (rvvergara) [inline assets/script/library.js]: Let us make sure we follow the correct way of including a `toggle` functionality for our book instances. This is through including a method in `Book`'s prototype:

image

```javascript
Book.prototype.toggleRead = function(){
  // logic here without any DOM related code
}
```

File: assets/script/library.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    return books;
+}
+
+
+function addBook(book) {
+    let books = getBooks();
+    books.push(book);
+    localStorage.books = JSON.stringify(books);
```
Comment (rvvergara) [inline assets/script/library.js]: To separate our DOM and localstorage related logic from our object logic, I propose we create a constructor/object creator for the library:

```javascript
function Library = () => {
  this.books = [ ];
}
// to add a new book:
Library.prototype.addBook = (book) => {
     this.books.push(book);
} 

Library.prototype.removeBook = (title) => {
  // assuming title is unique
  const indexToRemove = this.books.findIndex(book => book.title === title);
  this.books.splice(indexToRemove, 1);
}

const library = new Library();

// Event handler for our form submission:

function submitBook(library, dataFromForm){
  // create new instance of book from data coming from form
  const book = new Book(dataFromForm.title, dataFromForm.pages, dataFromForm.author);
 
// add this new book to the library
  library.addBook(book);

 // here below we should be calling another function that deals with localstorage logic
// assuming this is that function:
 recordToLocalStorage(/*necessary arguments*/);
}
```

**Review by rvvergara — CHANGES_REQUESTED** [review]
Hi @mosaaleb and @juzQrios,

Just some minor fixes I am proposing:

1. Let us prioritize the use of `const` rather than by default using `let`. We should use `let` on instances we are going to re-assign the variable to another value -> Best practices

2. Kindly see my comment on how to add the toggle functionality for our book instances.

3. Whenever we submit a new book, it would be better user-experience wise if the browser didn't refresh. Kindly fix this.

4. Our code mixes a lot of the DOM related logic with object related code. Let's make sure we separate them. Kindly see my suggested architecture for the library itself.

File: assets/script/library.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
+//--------------------------CLASS CODE-----------------------//
+class Book {
+    constructor(title, author, pagesNumber, isRead = false) {
+      this.title = title;
+      this.author = author;
+      this.pagesNumber = pagesNumber;
+      this.isRead = isRead; 
+    }
+    toggleRead() {
```
Comment (juzQrios) [inline assets/script/library.js]: Resolves https://github.com/mosaaleb/books/pull/1#discussion_r310853663

File: assets/script/library.js (hunk 1 above)
Comment (darshanjayadev) [reply assets/script/library.js]: Refactored code to include `toggleRead` method in Book prototype

File: assets/script/library.js (hunk 2 above)
Comment (darshanjayadev) [reply assets/script/library.js]: Added `Library` class and refactored existing code with modifications

**Review by rvvergara — CHANGES_REQUESTED** [review]
### Awesome refactor guys!! Just a little bit more. I promise this is to make our code base robust

1. Let us just sweep the whole `library.js` to ensure we are prioritizing the use of `const` vs `let`. Haha this is just a small pet peeve but will help you later on. It's a common habit we build when we first use ES6 (to use `let` by default). Not a big deal and certainly not a requirement to pass the project

2. I appreciate the use of `class` for `Book` and `Library` -> good job!!! However, I think probably it isn't the best implementation for the `UserInterface`. Why? We normally use constructors (which is present in a `class` and or a simple function constructor) if we are going to make multiple "instances" of something AND/OR the object we are creating the `class` for is a "unit entity" (my own terminology). For instance, `Library`, while having only a single instance for this app, is a "unit entity" -> the meaning of this will be clearer once we get into the "testing sections". `Book` is also a "unit" [now removing the word entity for greater familiarity with the context], so we can rightfully use class. However, `UserInterface` is more of like an "all-encompassing" thing in the project. Think about it (using as mental reference, RSpec) isn't it easy to test the `Library` and `Book` classes? I mean you can easily write a test for the constructor, the methods, etc. But how about writing a test for `UserInterface` ? :) 

I suggest we keep the several methods of `UserInterface` as separate functions. And let our event listeners attached to the different elements of our interface call the necessary methods of our `Book` and `Library`. I hope how I explained it is clear.

**Review by dipto0321 — APPROVED** [review]
HI @mosaaleb and @juzQrios ,

So far so good 👏 . Just one thing about UI. Let's hide the add book form every time when I add a new book. It should only appear when I/ a user clicked on Add new book button. After adding this feature you can merge this project. This is from me 👍 .

Thanks 🙏

**Pull request merged**